internal/ingest/download_sip.go

```diff
@@ -0,0 +1,97 @@
+package ingest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/google/uuid"
+	"gocloud.dev/gcerrors"
+
+	goaingest "github.com/artefactual-sdps/enduro/internal/api/gen/ingest"
+	"github.com/artefactual-sdps/enduro/internal/datatypes"
+	"github.com/artefactual-sdps/enduro/internal/persistence"
+)
+
+func (w *goaWrapper) checkSIP(ctx context.Context, id string) (*datatypes.SIP, error) {
```

Comment:
Maybe `isSIPDownloadable()` would provide a better hint at what this method is doing?

Reply:
I think it's hard to name this method because it's doing two things: loading the SIP and then checking if it's downloadable.  I think splitting up the method into a `loadSIP()` (or something similar) method and a `isFailed(s datatypes.SIP) bool` function would make naming easier and follow the "functions should do one thing" principle, even if it does require a few more lines of code.